tensorflow/compiler/mlir/hlo/lib/Transforms/buffer_packing.cc

```diff
@@ -366,8 +366,9 @@ class BufferPacking : BufferPlacementTransformationBase {
       // Find alloc position operation.
       mlir::OpBuilder packBuilder(&(block->front()));
       auto location = block->front().getLoc();
-      auto memrefType = MemRefType::get({packedBuffer.numSegments},
-                                        packBuilder.getIntegerType(8));
+      auto memrefType =
+          MemRefType::get({static_cast<long long>(packedBuffer.numSegments)},
```

Comment:
This should be a `int64_t`. The issue that arises here is that `numSegments` is `size_t` and `MemRefType::get` expects an `ArrayRef` of `int64_t`.